### DISCUSSION TITLE
Option to suspend blur (or suspend transperancy) for apps that are not focused to save on resources.

### DISCUSSION THREAD

**Discussion by Aakarshit-Sharma19** [tsujan/Kvantum#732] [Ideas]
So, Is such a feature possible where the inactive windows suspends its blur and only the window that is in focus should be blurred (Similar to uwp apps in windows 10 that have blur). When opening like 5 dolphin windows, konsole, okular simultaneously the whole ui lags during animations, or resizing and such. It might be the issue with buggy gpu drivers or unoptimized implementation of kwin blur, but this seriously hampers the kde experience. So, is such an option possible for kvantum and kwin.

**Comment by tsujan** [comment]
> .... unoptimized implementation of kwin blur,....

KWin has the best and most resource-friendly blur effect I've ever seen. I haven't encountered a problem with several translucent windows but some users have reported problems with only 2 windows. There are factors other than KWin too. I use Intel with modesetting and the default settings.

> Is such a feature possible where the inactive windows suspends its blur and only the window that is in focus should be blurred

It should be possible if the needed code is added but I'm not sure whether enabling the blur effect on activation and disabling it on deactivation will be resource-friendly. I'll experiment with it when I find the time and will tell you about the result here. Thanks for the idea!

**Comment by tsujan** [comment]
I found some time to write a part of the code for testing. There was a problem with disabling KWin's blur effect (sometimes artifacts appeared on windows without blurring) but I found a simple workaround.

I should test it more and complete the code before adding a commit but, so far, it seems acceptable: I saw to extra resource usage. I left the translucent windows that didn't accept focus to be always blurred.

There seems to be only one problem that's out of Kvantum's control: KWin's decorations are always blurred with KWin themes that support blurring. Understandably, it isn't our problem.

**Comment by tsujan** [comment]
Done in https://github.com/tsujan/Kvantum/commit/8cc949be14417af2372ff778d8be10bd4749da3c

The new option is in Kvantum Manager → Configure Active Theme → Hacks → No blurring for inactive windows.

**Reply by Aakarshit-Sharma19** [reply]
Thanks @tsujan, this works perfectly.

**Reply by tsujan** [reply]
Does it solve the problem you had with multiple windows? I ask because the problem doesn't happen here.

**Reply by Aakarshit-Sharma19** [reply]
Yes, there is no performance drop or lag whatsoever, with multiple windows as only one window is being blurred at a time. Too much to ask, but in next kvantum release, suspending blur + translucency would be a great feature which would cover up for the performance issues as well would look great.

**Reply by tsujan** [reply]
Good to know that there's no performance drop due to blurring. That may be helpful to many.

As for disabling translucency for inactive windows, it actually occurred to me but it's more complicated and I'm not sure if it's possible without introducing bugs. That being said, I'll think about it later.

I want to release Kvantum 1.0.0 before 2022. Because of that, I'll need to stop adding features or experimenting with new options for a while.
